templates/storypages/porsche-911-order/porsche-911-story.js

```diff
@@ -0,0 +1,721 @@
+    });
+  }
+
+  var nodeIcon = L.icon({
+    iconUrl: "assets/city-marker.svg",
+    iconSize: [28, 28],
+    iconAnchor: [14, 28],
+    popupAnchor: [0, -24]
+  });
+
+  var nodesLayer = L.geoJSON(null, {
+    pointToLayer: function(feature, latlng) {
+      return L.marker(latlng, { icon: nodeIcon });
+    },
+    onEachFeature: function(feature, layer) {
+      var props = feature.properties || {};
+      var lines = [];
+      if (props.name) lines.push('<strong>' + props.name + '</strong>');
+      if (props.city || props.country) {
+        var parts = [];
+        if (props.city) parts.push(props.city);
+        if (props.country) parts.push(props.country);
+        if (parts.length) lines.push('<small>' + parts.join(', ') + '</small>');
+      }
+      if (props.step) lines.push('<em>' + props.step + '</em>');
+      if (props.notes) lines.push(props.notes);
+      if (props.source_url) lines.push('<a href="' + props.source_url + '" target="_blank" rel="noopener">Source</a>');
+      var html = lines.join('<br />');
+      if (html) layer.bindPopup(html);
+    }
+  }).addTo(map);
+
+  var ROUTE_STYLES = {
+    information: { color: "#577590", dashArray: "6,6", weight: 2 },
+    supply: { color: "#588157", dashArray: "4,6", weight: 2 },
+    road: { color: "#e36414", weight: 3 },
+    rail: { color: "#e9c46a", dashArray: "4,4", weight: 3 },
+    sea: { color: "#277da1", weight: 3 },
+    "road/rail": { color: "#ca6702", dashArray: "4,6", weight: 3 },
+    default: { color: "#6c757d", weight: 2 }
+  };
+
+  var routesLayer = L.geoJSON(null, {
+    style: function(feature) {
+      var key = feature.properties && feature.properties.mode;
+      return ROUTE_STYLES[key] || ROUTE_STYLES.default;
+    },
+    onEachFeature: function(feature, layer) {
+      var props = feature.properties || {};
+      var lines = [];
+      if (props.name) lines.push('<strong>' + props.name + '</strong>');
+      if (props.notes) lines.push(props.notes);
+      var html = lines.join('<br />');
+      if (html) layer.bindPopup(html);
+    }
+  }).addTo(map);
+
+  var cityLayer = L.geoJSON(null, {
+    pane: "city-pane",
+    style: {
+      color: "#278ea5",
+      weight: 1.25,
+      fillColor: "rgba(39, 142, 165, 0.22)",
+      fillOpacity: 1
+    },
+    onEachFeature: function(feature, layer) {
+      var name = feature.properties && feature.properties.name;
+      var info = CITY_INFO[name];
+      layer.on("click", function(evt) {
+        stopLeafletClick(evt);
+        openInfoPanel(info ? info.title : (name || "City"), formatInfo(info));
+      });
+      if (name) layer.bindTooltip(name, { direction: "center", permanent: false });
+    }
+  }).addTo(map);
+
+  var countryLayer = null;
+  var regionLayer = null;
+  var allNodes = null;
+  var allRoutes = null;
+
+  function loadJson(url) {
+    return fetch(url).then(function(resp) {
+      if (!resp.ok) throw new Error('Failed to load ' + url);
+      return resp.json();
+    });
+  }
```

Comment:
Here's the exact same function with `async`/`await`. Which style you choose is mostly a matter of aesthetic preference.
```suggestion
  async function loadJson(url) {
    const resp = await fetch(url);
    if (!resp.ok) throw new Error('Failed to load ' + url);
    return resp.json();
  }
```